### PR TITLE
Ensure specs always see the host compiler's CRYSTAL_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by jhass**
Redux #9421

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,6 +37,7 @@ EXPORTS := \
   CRYSTAL_CONFIG_LIBRARY_PATH="$(shell crystal env CRYSTAL_LIBRARY_PATH)" \
   CRYSTAL_CONFIG_BUILD_COMMIT="$(CRYSTAL_CONFIG_BUILD_COMMIT)" \
 	SOURCE_DATE_EPOCH="$(SOURCE_DATE_EPOCH)"
+SPEC_EXPORTS := CRYSTAL_LIBRARY_PATH="$(shell crystal env CRYSTAL_LIBRARY_PATH)"
```
Comment (bcardiff) [inline Makefile:40]: Any second thought on https://github.com/crystal-lang/crystal/pull/9423/files#r435373542 ?

Comment (jhass) [reply Makefile:40]: Ah, no. I just copy pasted here because I was counting on having to rebase on #9423 anyways :)

File: bin/crystal [hunk 2]
```diff
@@ -141,6 +141,10 @@ CRYSTAL_DIR="$CRYSTAL_ROOT/.build"
 export CRYSTAL_PATH=lib:$CRYSTAL_ROOT/src
 export CRYSTAL_HAS_WRAPPER=true
 
+if [ -z "$CRYSTAL_LIBRARY_PATH" ]; then
+  export CRYSTAL_LIBRARY_PATH="$(crystal env CRYSTAL_LIBRARY_PATH || echo "")"
```
Comment (bcardiff) [inline bin/crystal:145]: Why is this needed?

If .build/crystal does not exist, the crystal binary will have the CRYSTAL_LIBRARY_PATH
If .build/crystal exists, the Makefile should have added the CRYSTAL_CONFIG_LIBRARY_PATH

Comment (jhass) [reply bin/crystal:145]: for running individual specs with `bin/crystal spec`.  The compiled spec binary does not inherit `CRYSTAL_CONFIG_LIBRARY_PATH`,  thus if there's a `run` helper call in it, it invokes the linker without the necessary arguments.

Comment (bcardiff) [reply bin/crystal:145]: Ok, got it.

I think is better to set `CRYSTAL_CONFIG_LIBRARY_PATH`. This way the wrapper is able to build a compiler on demand for this scenario, but we don't change the CRYSTAL_LIBRARY_PATH for every run. Since you might want to create a .build/crystal with some specific CRYSTAL_LIBRARY_PATH and compile programs with it.

```
if [ -z "$CRYSTAL_CONFIG_LIBRARY_PATH" ]; then
  export CRYSTAL_CONFIG_LIBRARY_PATH="$(crystal env CRYSTAL_LIBRARY_PATH || echo "")"
fi
```

File: Makefile (hunk 1 above)
Comment (bcardiff) [reply Makefile:40]: While doing #9423 I consider adding $(EXPORTS) to $(O)/all_spec, $(O)/compiler_spec.

CRYSTAL_LIBRARY_PATH should not be needed in std_spec (as long as we don't build the compiler) and the installed compiler in PATH or .build/crystal has that set.

Would that work? I can add those changes in #9423

File: bin/crystal (hunk 2 above)
Comment (bcardiff) [reply bin/crystal:145]: I can add that in #9423 if you agree. The changes are all related.

Comment (jhass) [reply bin/crystal:145]: Sure, let's condense this into #9423 :)

File: Makefile (hunk 1 above)
Comment (jhass) [reply Makefile:40]: I considered this too, but wasn't too sure because it sets so many. But I guess it does no harm :)
